packages/block-library/src/tag-cloud/edit.js

```diff
@@ -53,10 +53,14 @@ function TagCloudEdit( { attributes, setAttributes } ) {
 	const units = useCustomUnits( {
 		availableUnits: availableUnits || [ '%', 'px', 'em', 'rem' ],
 	} );
-	const taxonomies = useSelect(
-		( select ) => select( coreStore ).getTaxonomies( { per_page: -1 } ),
-		[]
-	);
+	const taxonomies = useSelect( ( select ) => {
+		const _taxonomies = select( coreStore ).getTaxonomies( {
+			per_page: -1,
+		} );
+		return _taxonomies?.filter(
+			( { visibility } ) => visibility?.publicly_queryable
+		);
```

Comment:
We should avoid using `filter`, `map`, and similar actions inside the selector. Plus, `getTaxonomyOptions` already filters and maps values.

Reply:
Yeah, `show_tagcloud` would do it.